### PR TITLE
refactor: Migrate DebugUtil to TypeScript

### PR DESCRIPTION
**Pull request by bennyn**
I migrated the debug util to TypeScript and removed unused functions from it.

### CODE REVIEW

File: src/script/util/DebugUtil.ts [hunk 1]
```diff
@@ -0,0 +1,319 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (lipis) [inline src/script/util/DebugUtil.ts:3]: since it's a new file maybe 2020?

File: src/script/storage/StorageService.ts [hunk 2]
```diff
@@ -50,7 +50,8 @@ enum DEXIE_CRUD_EVENT {
 }
 
 export class StorageService {
-  public db?: Dexie & DexieObservable;
+  // Quickfix table name index until proper db typing: https://dexie.org/docs/Typescript#create-a-subclass
+  public db?: Dexie & DexieObservable & {[tableName: string]: any};
```
Comment (lipis) [inline src/script/storage/StorageService.ts:54]: any? :)

**Review by lipis — APPROVED** [review]

File: src/script/storage/StorageService.ts (hunk 2 above)
Comment (bennycode) [reply src/script/storage/StorageService.ts:54]: See above: Quickfix

File: src/script/util/DebugUtil.ts [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+          .getNotifications(undefined, undefined, EventRepository.CONFIG.NOTIFICATION_BATCHES.MAX)
+          .then(({notifications}: NotificationList) => ({
+            message,
+            notifications,
+          }));
+      })
+      .then(({message, notifications}: {message: ContentMessage; notifications: Notification[]}) => {
+        const dateTime = new Date(message.timestamp());
+        return notifications
+          .reduce((accumulator, notification) => accumulator.concat(notification.payload), [])
```
Comment (AndyLnd) [inline src/script/util/DebugUtil.ts]:
```suggestion
          .flatMap(({payload}) => payload)
```

Comment (AndyLnd) [reply src/script/util/DebugUtil.ts]: As far as I can tell, our TS config supports `flatMap`

**Review by AndyLnd — APPROVED** [review]

File: src/script/util/DebugUtil.ts (hunk 1 above)
Comment (bennycode) [reply src/script/util/DebugUtil.ts:3]: It's not really a new file but a renamed one. Copyright on the code still stays the same.

**Pull request merged**